.azure-pipelines/templates/codeQualityChecks.yml

```diff
@@ -0,0 +1,71 @@
+stages:
+  - stage: 'code_quality'
+    displayName: 'Code Quality Checks'
+    jobs:
+      - job: 'cpp_checks'
+        displayName: 'C++ Checks'
+        pool:
+          vmImage: 'ubuntu-latest'
+        steps:
+          - bash: |
+              set -ex
+              wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+              sudo add-apt-repository --yes 'deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic main'
+              sudo apt-get update
+              sudo apt-get install --yes clang-format-7
+            displayName: 'Install tooling'
+          - bash: |
+              set -ex
+              clang-format-7 --version
+              find . -type f -regex '.*\.\(c\|cpp\|h\|hpp\|hui\)' -exec clang-format-7 -i {} +
+              git diff
+              git diff --quiet
+            displayName: 'Run clang-format'
+      - job: 'python_checks'
+        displayName: 'Python Checks'
+        pool:
+          vmImage: 'ubuntu-latest'
+        steps:
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: '3.x'
+              addToPath: true
+              architecture: 'x64'
+          - bash: |
+              set -ex
+              pip install flake8
+              python -m flake8 .
+            displayName: 'Run pep8-format'
+      - job: 'cmake_checks'
+        displayName: 'CMake Checks'
+        pool:
+          vmImage: 'ubuntu-latest'
+        steps:
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: '3.x'
+              addToPath: true
+              architecture: 'x64'
+          - bash: pip install cmake_format==0.6.9
+            displayName: 'Install tooling'
+          - bash: |
+              set -ex
+              cmake-format --version
+              find . -type f -name CMakeLists.txt -exec cmake-format -i '{}' \;
+              git diff
+              git diff --quiet
+            displayName: 'Run cmake-format'
+      - job: 'copyright_year'
+        displayName: 'Copyright Year'
+        pool:
+          vmImage: 'ubuntu-latest'
+        steps:
+          - bash: |
+              set -ex
+              # Find all occurances of "YYYY Team Dissolve and contributors" and replace YYYY with the current year
+              find . -type f -not -path '.git/*' -exec sed -ri "s/[0-9]{4} (Team Dissolve and contributors)/$(date +%Y) \1/g" {} +
```

Comment:
This should match the text in the SPDF copyright header - e.g. `2021 E. Devlin and T. Youngs`.